### DISCUSSION TITLE
v2 to v3 Upgrade: Confirm One Less Timestep Written

### DISCUSSION THREAD

**Discussion by jamespolly** [Parcels-code/Parcels#1542] [Q&A] [answered]
Hello! I'm upgrading the version of parcels in my codebase from 2.4.x to 3.0.0.

I'd like to confirm that this upgrade has the following effect for the same simulation of two particles over a 5 hour runtime with a one hour output_dt. Below are the `.sizes` attributes of the zarr stores opened as xarray Datasets. I get one more output time (`obs`) in the v2 output.
```
In [14]: op_v2.sizes
Out[14]: Frozen({'trajectory': 2, 'obs': 6})

In [15]: op_v3.sizes
Out[15]: Frozen({'trajectory': 2, 'obs': 5})
```

Based on some cursory review, this part of the documentation seems most relevant: https://docs.oceanparcels.org/en/latest/examples/tutorial_kernelloop.html#3.-The-last-time-is-not-written-to-file

If someone can confirm that this is a known change in the behavior in v3.0.0 that would be most appreciated!

**Comment by erikvansebille** [comment, marked as answer]
Yep, this is correct and expected behaviour. If you still want the last output written too, you can use `pfile.write_latest_locations(pset, time=endtime)` (but note that this may not be correct values for variables other than longitude, latitude and depth). 

Alternative is to let your run go on for a little bit longer (i.e. set your `runtime` or `endtime` a bit larger)

**Reply by jamespolly** [reply]
Erik, thank you for the timely confirmation, and obviously for all that's going on here in `parcels`.
